### PR TITLE
[codex] Extract Supabase session helpers

### PR DESCRIPTION
**Pull request by LionSR**
## Summary
- move the Supabase stored-session schema, parser, and native session conversion into host-neutral auth helpers
- move auth callback token parsing out of the VS Code auth provider while preserving fragment/query precedence
- add focused tests for stored sessions, native-session conversion, and callback token edge cases

Stacked on #3284 / codex/electron-supabase-token-provider for PRD #14.

## Validation
- npm run format
- npm run compile:safe
- npm run compile
- npm run lint
- npm run compile-tests
- npx mocha --require ./out/test/setup.js out/test/auth/SupabaseSession.test.js out/test/auth/SupabaseClient.test.js
- git diff --check

---

> [!NOTE]
> **Medium Risk**
> Touches authentication session parsing and OAuth/magic-link callback token extraction, where small precedence/validation differences can break sign-in flows. Changes are mostly a refactor with added tests, reducing but not eliminating rollout risk.
> 
> **Overview**
> Centralizes Supabase session handling by moving the stored-session Zod schema, SecretStorage parsing, and native Supabase `Session` conversion into new `SupabaseSession` helpers, and updates `SupabaseAuthProvider` to use them.
> 
> Extracts OAuth/magic-link callback token parsing into `parseAuthCallbackTokens` (fragment-first with query fallback) and adds focused unit tests for stored-session parsing, native-session conversion defaults, and callback edge cases. Also factors `toErrorMessage` into `errorMessage.ts` and re-exports it from `errorHandlingUtils`, and re-exports the default session expiry constant via `config.ts`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 1a3f147c5ebbbd791067cd63702e25d8cd9d70a3. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/auth/SupabaseSession.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+ * Tokens usually arrive in the fragment, with query params as a fallback for
+ * PKCE and web environments.
+ */
+export function parseAuthCallbackTokens(
+  parts: AuthCallbackParts,
+): AuthCallbackTokenResult {
+  const fragmentParams = new URLSearchParams(parts.fragment);
+  const queryParams = new URLSearchParams(parts.query);
+  const getParam = (name: string): string | null =>
+    fragmentParams.get(name) ?? queryParams.get(name);
```
[the remark this answers is not in the document]
Comment (LionSR) [reply src/auth/SupabaseSession.ts:111]: Keeping `??` here is intentional. Empty fragment token parameters should be treated as authoritative rather than falling back to query values; this preserves the fragment-precedence behavior from the auth-callback fix and is covered by `does not fall back to query tokens when fragment tokens are empty` in `SupabaseSession.test.ts`.

File: src/auth/SupabaseSession.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
+import { z } from 'zod';
+import type { Session as SupabaseNativeSession } from '@supabase/supabase-js';
+
+export const DEFAULT_SUPABASE_SESSION_EXPIRY_MS = 60 * 60 * 1000;
```
[the remark this answers is not in the document]
Comment (LionSR) [reply src/auth/SupabaseSession.ts:5]: Addressed in `1a3f147c5`: `config.ts` now re-exports `DEFAULT_SUPABASE_SESSION_EXPIRY_MS` as `DEFAULT_SESSION_EXPIRY_MS`, so there is only one concrete expiry value and the provider/storage helpers share it.
